src/api/api_client.rs

```diff
@@ -179,29 +191,31 @@ where
 		self
 	}
 
-	fn _get_genesis_hash(client: &Client) -> ApiResult<Hash> {
+	/// Gets genesis hash directly from the node. Not a public method, because once the Api is created,
+	/// the genesis hash can be retrieved directly from the Api.
+	pub fn get_genesis_hash_from_node(client: &Client) -> ApiResult<Hash> {
 		let jsonreq = json_req::chain_get_genesis_hash();
-		let genesis = Self::_get_request(client, jsonreq)?;
+		let genesis = client.get_request(jsonreq)?;
 
 		match genesis {
 			Some(g) => Hash::from_hex(g).map_err(|e| e.into()),
 			None => Err(ApiClientError::Genesis),
 		}
 	}
 
-	fn _get_runtime_version(client: &Client) -> ApiResult<RuntimeVersion> {
+	pub fn get_runtime_version_from_node(client: &Client) -> ApiResult<RuntimeVersion> {
 		let jsonreq = json_req::state_get_runtime_version();
-		let version = Self::_get_request(client, jsonreq)?;
+		let version = client.get_request(jsonreq)?;
 
 		match version {
 			Some(v) => serde_json::from_str(&v).map_err(|e| e.into()),
 			None => Err(ApiClientError::RuntimeVersion),
 		}
 	}
 
-	fn _get_metadata(client: &Client) -> ApiResult<RuntimeMetadataPrefixed> {
+	pub fn get_metadata_from_node(client: &Client) -> ApiResult<RuntimeMetadataPrefixed> {
 		let jsonreq = json_req::state_get_metadata();
```

Comment:
The names `get_metadata`, `get_metadata_from_node` (and similar ones) are a bit confusing, as if the metadata returned by `get_metadata` is not the node's metadata. The difference is not the source of the metadata, but its currentness. 
Wouldn't it be better to have `get_cached_metadata` for `get_metadata` and let `get_metadata` for `get_metadata_from_node` ?

Reply:
I see your point, but I think the `get_metadata` should stay as is, because that should be the usual way to retrieve the metadata. Getting the metadata from the node should be the non-normal way, therefore warranting the longer name, I believe.

How about `query_metadata_from_node`?

Reply:
Perfect. That's clear. Thanks

Reply:
I think we are mixing concepts here, `get_*` sometimes performs a network request now, and sometimes it does not, this inconsistency is confusing.

I suggest `get_*` always talks to the node, then we don't need to distinguish between `query_metadata_from_node` to emphasize that it actually does the same as most of the `get_*` calls.

We don't have to re-invent the wheel here, we can just do:

`api.metadata()` is a simple getter to the field
`api.get_metdata()` is the network request, however, do we even want to keep it?

Same for the runtime version.

Reply:
> api.get_metdata() is the network request, however, do we even want to keep it?

Not sure. We need a private node query method, because the api needs it to be able to update the cached metdata, but it does not need to be public.. because one could just run `update_...` and then query the cached metadata.

I don't have a strong opinion here, but well.. why not make it private and force the user to update cached metadata if up-to-date metadata is needed.

Reply:
> I don't have a strong opinion here, but well.. why not make it private and force the user to update cached metadata if up-to-date metadata is needed.

This seems good to me! However, the private one should still be `get_metadata` IMO. :)

Reply:
Renamed accordingly 👍

Reply:
Also restructured a little, because it was veery chaotic. Updated the PR description accordingly to what has been changed.